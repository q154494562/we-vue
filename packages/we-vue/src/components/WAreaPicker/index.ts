import WAreaPicker from './WAreaPicker'

export { WAreaPicker }
export default WAreaPicker
