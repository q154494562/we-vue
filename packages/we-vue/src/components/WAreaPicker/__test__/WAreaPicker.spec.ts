import { mount } from '@vue/test-utils'
import WAreaPicker from '../'

import areaList from './area'

describe('area picker', () => {
  test('create', () => {
    const wrapper = mount(WAreaPicker)

    expect(wrapper.name()).toBe('w-area-picker')

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('render with area list', () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('watch value', async () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
      },
    })

    const spy = jest.spyOn(wrapper.vm, 'setOptions')

    await wrapper.vm.$nextTick()

    wrapper.setProps({
      value: '110000',
    })

    expect(spy).toHaveBeenCalled()

    expect(wrapper.vm.internalValue).toBe('110000')
  })

  test('watch areaList', async () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {},
    })

    await wrapper.vm.$nextTick()

    const spy = jest.spyOn(wrapper.vm, 'setOptions')

    wrapper.setProps({
      areaList: areaList,
    })

    expect(spy).toHaveBeenCalled()
  })

  test('reset method', async () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
      },
    })

    await wrapper.vm.$nextTick()

    const spy = jest.spyOn(wrapper.vm, 'setOptions')

    wrapper.vm.reset()

    expect(spy).toHaveBeenCalled()
  })

  test('getValues', async () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
        value: '110000',
      },
    })

    await wrapper.vm.$nextTick()

    expect(wrapper.vm.getArea()).toEqual({
      code: '110000',
      country: '',
      city: '北京市',
      county: '东城区',
      province: '北京市',
    })
  })

  test('onChange method', () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
      },
    })

    const values = [
      {
        name: 'foo',
        code: 'foo',
      },
      {
        name: 'bar',
        code: 'bar',
      },
      {
        name: 'baz',
        code: 'baz',
      },
    ]

    wrapper.vm.onChange(wrapper.vm.$refs.picker, values, 0)

    expect(wrapper.vm.internalValue).toBe('foo')
    expect(wrapper.emitted('change')).toBeTruthy()
  })

  test('click cancel button', () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
      },
    })

    wrapper.findAll('.weui-picker__action').at(0).trigger('click')

    expect(wrapper.vm.isActive).toBe(false)
    expect(wrapper.emitted('cancel')).toBeTruthy()
  })

  // TODO
  test('click confirm button', () => {
    const wrapper = mount(WAreaPicker, {
      propsData: {
        areaList: areaList,
        value: '110000',
      },
    })

    wrapper.findAll('.weui-picker__action').at(1).trigger('click')

    expect(wrapper.vm.isActive).toBe(false)
    expect(wrapper.emitted('confirm')).toBeTruthy()
  })
})
