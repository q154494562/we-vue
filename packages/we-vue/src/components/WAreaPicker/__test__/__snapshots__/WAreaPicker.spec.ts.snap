// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`area picker create 1`] = `
<div>
  <div class="weui-mask" style="display: none;"></div>
  <div class="weui-picker" style="display: none;">
    <div class="weui-picker__hd">
      <div class="weui-picker__action">取消</div>
      <div class="weui-picker__action">确定</div>
    </div>
    <div class="weui-picker__bd" style="height: 238px;">
      <div class="weui-picker__group">
        <div class="weui-picker__mask" style="background-size: 100% 102px;"></div>
        <div class="weui-picker__indicator" style="top: 102px;"></div>
        <div class="weui-picker__content" style="transform: translate3d(0, 102px, 0);"></div>
      </div>
      <div class="weui-picker__group">
        <div class="weui-picker__mask" style="background-size: 100% 102px;"></div>
        <div class="weui-picker__indicator" style="top: 102px;"></div>
        <div class="weui-picker__content" style="transform: translate3d(0, 102px, 0);"></div>
      </div>
      <div class="weui-picker__group">
        <div class="weui-picker__mask" style="background-size: 100% 102px;"></div>
        <div class="weui-picker__indicator" style="top: 102px;"></div>
        <div class="weui-picker__content" style="transform: translate3d(0, 102px, 0);"></div>
      </div>
    </div>
  </div>
</div>
`;

exports[`area picker render with area list 1`] = `
<div>
  <div class="weui-mask" style="display: none;"></div>
  <div class="weui-picker" style="display: none;">
    <div class="weui-picker__hd">
      <div class="weui-picker__action">取消</div>
      <div class="weui-picker__action">确定</div>
    </div>
    <div class="weui-picker__bd" style="height: 238px;">
      <div class="weui-picker__group">
        <div class="weui-picker__mask" style="background-size: 100% 102px;"></div>
        <div class="weui-picker__indicator" style="top: 102px;"></div>
        <div class="weui-picker__content" style="transform: translate3d(0, 102px, 0);">
          <div class="weui-picker__item">北京市</div>
          <div class="weui-picker__item">天津市</div>
          <div class="weui-picker__item">河北省</div>
          <div class="weui-picker__item">山西省</div>
          <div class="weui-picker__item">内蒙古自治区</div>
          <div class="weui-picker__item">辽宁省</div>
          <div class="weui-picker__item">吉林省</div>
          <div class="weui-picker__item">黑龙江省</div>
          <div class="weui-picker__item">上海市</div>
          <div class="weui-picker__item">江苏省</div>
          <div class="weui-picker__item">浙江省</div>
          <div class="weui-picker__item">安徽省</div>
          <div class="weui-picker__item">福建省</div>
          <div class="weui-picker__item">江西省</div>
          <div class="weui-picker__item">山东省</div>
          <div class="weui-picker__item">河南省</div>
          <div class="weui-picker__item">湖北省</div>
          <div class="weui-picker__item">湖南省</div>
          <div class="weui-picker__item">广东省</div>
          <div class="weui-picker__item">广西壮族自治区</div>
          <div class="weui-picker__item">海南省</div>
          <div class="weui-picker__item">重庆市</div>
          <div class="weui-picker__item">四川省</div>
          <div class="weui-picker__item">贵州省</div>
          <div class="weui-picker__item">云南省</div>
          <div class="weui-picker__item">西藏自治区</div>
          <div class="weui-picker__item">陕西省</div>
          <div class="weui-picker__item">甘肃省</div>
          <div class="weui-picker__item">青海省</div>
          <div class="weui-picker__item">宁夏回族自治区</div>
          <div class="weui-picker__item">新疆维吾尔自治区</div>
          <div class="weui-picker__item">台湾省</div>
          <div class="weui-picker__item">香港特别行政区</div>
          <div class="weui-picker__item">澳门特别行政区</div>
        </div>
      </div>
      <div class="weui-picker__group">
        <div class="weui-picker__mask" style="background-size: 100% 102px;"></div>
        <div class="weui-picker__indicator" style="top: 102px;"></div>
        <div class="weui-picker__content" style="transform: translate3d(0, 102px, 0);">
          <div class="weui-picker__item">北京市</div>
        </div>
      </div>
      <div class="weui-picker__group">
        <div class="weui-picker__mask" style="background-size: 100% 102px;"></div>
        <div class="weui-picker__indicator" style="top: 102px;"></div>
        <div class="weui-picker__content" style="transform: translate3d(0, 102px, 0);">
          <div class="weui-picker__item">东城区</div>
          <div class="weui-picker__item">西城区</div>
          <div class="weui-picker__item">朝阳区</div>
          <div class="weui-picker__item">丰台区</div>
          <div class="weui-picker__item">石景山区</div>
          <div class="weui-picker__item">海淀区</div>
          <div class="weui-picker__item">门头沟区</div>
          <div class="weui-picker__item">房山区</div>
          <div class="weui-picker__item">通州区</div>
          <div class="weui-picker__item">顺义区</div>
          <div class="weui-picker__item">昌平区</div>
          <div class="weui-picker__item">大兴区</div>
          <div class="weui-picker__item">怀柔区</div>
          <div class="weui-picker__item">平谷区</div>
          <div class="weui-picker__item">密云区</div>
          <div class="weui-picker__item">延庆区</div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
